.github/workflows/tests-unpinned.yml

```diff
@@ -0,0 +1,34 @@
+name: Tests with unpinned dependencies
+
+on:
+  schedule:
+    - cron: 0 0 * * 6
+
+jobs:
+  test:
+    name: Run tests
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-14, windows-latest]
```

Comment:
What's the reason for `macos-14` as opposed to `macos-latest`? If we want forewarning on new versions / OSes, it seems counter-intuitive to pin the macos version here.

Reply:
Sorry should have put a comment explaining this and also adding a TODO to update to `macos-latest` in future. Currently `macos-latest` I think maps to an earlier MacOS version that importantly doesn't run on Apple Silicon, although this is due to change in the next few months- see https://github.com/orgs/community/discussions/102846 and https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/ . I think testing against MacOS on M1 would be of the most value as we seem to have been hitting problems with dependencies most often on Apple Silicon machines. Will add an explanatory comment with a link to the discussion and a note to update to `macos-latest` once this is update to point to `macos-14` or later

Reply:
Hmm actually from https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories it looks like this has now been updated so looks like can just move directly to `macos-latest`